src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> oldLast = last;
+        Node<T> newLast = new Node<>(oldLast, value, null);
+        last = newLast;
+        if (oldLast == null) {
+            first = newLast;
+        } else {
+            oldLast.next = newLast;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexPossibility(index);
```

Comment:
if you firstly check index == size than you can use checkIndex checkIndexExistence)

Reply:
in my first commit that was the way I realized checks. aleksandr-bevza told me to mix those methods together

Reply:
And I'm writing you the same, but you left the two methods)